src/Voron/StorageEnvironment.cs

```diff
@@ -871,7 +871,11 @@ public void FlushLogToDataFile()
 
         internal void AssertNoCatastrophicFailure()
         {
-            CatastrophicFailure?.Throw(); // force re-throw of error
+            if (CatastrophicFailure == null)
+                return;
+
+            _options.CatastrophicFailureNotification.RaiseNotification(CatastrophicFailure.SourceException);
```

Comment:
Won't this be called mulitple times when we reached a failure? Then after 2 seconds, we'll unload?